File: WappChatAnalyzer-frontend/src/app/dtos/statistic.ts
import { Sender } from "./sender";

export class Statistic {
    public statisticName: string;

    public filter: { fromDate: Date, toDate: Date, groupingPeriod: string };

    public senders: { [Key: number]: Sender }

    public total: number;
    public totalBySenders: { [Key: number]: number };

    public timePeriods: string[];
    public valuesBySendersOnTimePeriods: { [Key: number]: number[]; };
    public totalsOnTimePeriods: number[];
}

File: WappChatAnalyzer/DTOs/Statistic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WappChatAnalyzer.DTOs
{
    public class Statistic
    {
        public string StatisticName { get; set; }

        public Filter Filter { get; set; }

        public Dictionary<int, SenderDTO> Senders { get; set; }

        public float Total { get; set; }
        public Dictionary<int, float> TotalBySenders { get; set; }

        public List<DateTime> TimePeriods { get; set; }
        public Dictionary<int, List<float>> ValuesBySendersOnTimePeriods { get; set; }
        public List<float> TotalsOnTimePeriods { get; set; }
    }
}
